tests/bdd.nim

```diff
@@ -0,0 +1,34 @@
+import std/sugar
+
+export sugar
+
+proc itShould*(
+  msg: string,
+  name = "test name: ",
+  cond: bool,
+  istrue = true
+  ): void = doAssert cond == istrue, name & " -> " & msg
+
+proc itShouldNot*(
+  msg: string,
+  name = "test name: ",
+  cond: bool
+  ): void = itShould msg, name, cond, false
+
+# TODO: follow the leader: https://www.chaijs.com/api/bdd/
+type What* = enum
+  should, shouldNot, shouldError, shouldNotError
+
+proc bdd*(caseName: string): (What, string, () -> bool) -> void =
```

Comment:
add a bddAsync 
we'll need it for fetching remote files and other things down stream

Reply:
and it looks like theres a space after ()     -> bool

Reply:
nope thats just destiny winning the fight of old age